packages/server-renderer/src/render.ts

```diff
@@ -218,7 +218,10 @@ export function renderVNode(
   parentComponent: ComponentInternalInstance,
   slotScopeId?: string
 ) {
-  const { type, shapeFlag, children } = vnode
+  const { type, shapeFlag, children, dirs, props } = vnode
+  if (dirs) {
+    vnode.props = applySSRDirectives(vnode, props, dirs)
+  }
```

Comment:
I think this will lead to directives being called twice for normal elements.

I experimented with changing this test:

https://github.com/vuejs/core/blob/7947e3bcea0689d57db1e47af9b531cc93540e0c/packages/server-renderer/__tests__/ssrDirectives.spec.ts#L494

I added in some code to count how many times `getSSRProps` is called. It should be once, but with the code on this branch it gets called twice.

Reply:
fixed via https://github.com/vuejs/core/pull/7247/commits/d8b6e89d3aa4566dba9d10b400dbbd5f2353a4bd